drivers/i2c/i2c_ll_stm32.c

```diff
@@ -230,39 +230,20 @@ static int i2c_stm32_init(const struct device *dev)
 	}
 
 	if (clock_control_on(clk,
-		(clock_control_subsys_t *) &cfg->pclken) != 0) {
+		(clock_control_subsys_t *) &cfg->pclken[0]) != 0) {
 		LOG_ERR("i2c: failure enabling clock");
 		return -EIO;
 	}
 
-#if defined(CONFIG_SOC_SERIES_STM32F3X) || defined(CONFIG_SOC_SERIES_STM32F0X)
-	/*
-	 * STM32F0/3 I2C's independent clock source supports only
-	 * HSI and SYSCLK, not APB1. We force I2C clock source to SYSCLK.
-	 * I2C2 on STM32F0 uses APB1 clock as I2C clock source
-	 */
-
-	switch ((uint32_t)cfg->i2c) {
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(i2c1), okay)
-	case DT_REG_ADDR(DT_NODELABEL(i2c1)):
-		LL_RCC_SetI2CClockSource(LL_RCC_I2C1_CLKSOURCE_SYSCLK);
-		break;
-#endif
-
-#if defined(CONFIG_SOC_SERIES_STM32F3X) &&	\
-	DT_NODE_HAS_STATUS(DT_NODELABEL(i2c2), okay)
-	case DT_REG_ADDR(DT_NODELABEL(i2c2)):
-		LL_RCC_SetI2CClockSource(LL_RCC_I2C2_CLKSOURCE_SYSCLK);
-		break;
-#endif
-
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(i2c3), okay)
-	case DT_REG_ADDR(DT_NODELABEL(i2c3)):
-		LL_RCC_SetI2CClockSource(LL_RCC_I2C3_CLKSOURCE_SYSCLK);
-		break;
-#endif
+#if (DT_NUM_CLOCKS(DT_NODELABEL(name)) > 1)
```

Comment:
Contrary to `STM32_I2C_INIT()` case, `name` is undefined, this will likely not produce the expected result.